pkg/solver/server.go

```diff
@@ -62,17 +63,44 @@ func NewSolverServer(
  *
 */
 
+func exemptIPKeyFunc(exemptIPs []string) func(r *corehttp.Request) (string, error) {
+	return func(r *corehttp.Request) (string, error) {
+		ip, _, err := net.SplitHostPort(r.RemoteAddr)
+		if err != nil {
+			ip = r.RemoteAddr
+		}
+		
+
+		for _, exemptIP := range exemptIPs {
+			if ip == exemptIP {
+				return "", nil
+			}
+		}
+		
+		return ip, nil
+	}
+}
+
 func (solverServer *solverServer) ListenAndServe(ctx context.Context, cm *system.CleanupManager, tracerProvider *trace.TracerProvider) error {
 	router := mux.NewRouter()
 
 	subrouter := router.PathPrefix(http.API_SUB_PATH).Subrouter()
 
 	subrouter.Use(http.CorsMiddleware)
 	subrouter.Use(otelmux.Middleware("solver", otelmux.WithTracerProvider(tracerProvider)))
+
+
+	exemptIPs := solverServer.options.RateLimiter.ExemptedIPs
+
+	log.Debug().Strs("exemptIPs", exemptIPs).Msg("Loaded rate limit exemptions")
+
```

Comment:
What would you think about passing `solverServer.options.RateLimiter.ExemptedIPs` directly to `exemptIPKeyFunc` and logging the exempt IPs in there? Would clean things up a bit here.